olp-cpp-sdk-core/src/network2/curl/NetworkProtocolCurl.cpp

```diff
@@ -0,0 +1,1230 @@
+    CURL* handle, std::size_t retryCount) {
+  std::vector<std::pair<std::string, std::string> > statistics;
+  double time;
+  curl_easy_getinfo(handle, CURLINFO_TOTAL_TIME, &time);
+  statistics.push_back(
+      std::pair<std::string, std::string>("TotalTime", std::to_string(time)));
+  curl_easy_getinfo(handle, CURLINFO_NAMELOOKUP_TIME, &time);
+  statistics.push_back(std::pair<std::string, std::string>(
+      "NameLookupTime", std::to_string(time)));
+  curl_easy_getinfo(handle, CURLINFO_CONNECT_TIME, &time);
+  statistics.push_back(
+      std::pair<std::string, std::string>("ConnectTime", std::to_string(time)));
+  curl_easy_getinfo(handle, CURLINFO_APPCONNECT_TIME, &time);
+  statistics.push_back(std::pair<std::string, std::string>(
+      "AppConnectTime", std::to_string(time)));
+  curl_easy_getinfo(handle, CURLINFO_PRETRANSFER_TIME, &time);
+  statistics.push_back(std::pair<std::string, std::string>(
+      "PreTransferTime", std::to_string(time)));
+  curl_easy_getinfo(handle, CURLINFO_STARTTRANSFER_TIME, &time);
+  statistics.push_back(std::pair<std::string, std::string>(
+      "StartTransferTime", std::to_string(time)));
+  curl_easy_getinfo(handle, CURLINFO_REDIRECT_TIME, &time);
+  statistics.push_back(std::pair<std::string, std::string>(
+      "RedirectTime", std::to_string(time)));
+  statistics.push_back(std::pair<std::string, std::string>(
+      "Retries", std::to_string(retryCount)));
+  return statistics;
+}
+
+#ifdef NETWORK_HAS_OPENSSL
+bool file_exists(const std::string& file_path) {
+  if (file_path.empty()) {
+    return false;
+  }
+
+  struct stat stat_info;
+
+  return (0 == stat(file_path.c_str(), &stat_info)) &&
+         S_ISREG(stat_info.st_mode);
+}
+
+const auto curl_ca_bundle_name = "ca-bundle.crt";
+
+std::string default_ca_bundle_path() {
+  return curl_ca_bundle_name;
+}
+
+std::string alternative_ca_bundle_path() {
+  return curl_ca_bundle_name;
+}
+
+std::string ca_bundle_path() {
+  std::string bundle_path;
+  bundle_path = default_ca_bundle_path();
+  if (!file_exists(bundle_path)) {
+    bundle_path = alternative_ca_bundle_path();
+  }
+  if (!file_exists(bundle_path)) {
+    bundle_path.clear();
+  }
+  return bundle_path;
+}
+#endif
+
+// To avoid static_cast and possible values changes in CURL
+curl_proxytype to_curl_proxy_type(olp::network2::NetworkProxySettings::Type type) {
+  using ProxyType = olp::network2::NetworkProxySettings::Type;
+  switch (type) {
+    case ProxyType::HTTP:
+      return CURLPROXY_HTTP;
+    case ProxyType::SOCKS4:
+      return CURLPROXY_SOCKS4;
+    case ProxyType::SOCKS5:
+      return CURLPROXY_SOCKS5;
+    case ProxyType::SOCKS4A:
+      return CURLPROXY_SOCKS4A;
+    case ProxyType::SOCKS5_HOSTNAME:
+      return CURLPROXY_SOCKS5_HOSTNAME;
+    default:
+      assert(false);
+      return CURLPROXY_HTTP;
+  }
+}
+}  // namespace
+
+namespace olp {
+namespace network2 {
+const char* kLogTag = "CURL";
+
+namespace {
+int convertErrorCode(CURLcode curlCode) {
```

Comment:
```suggestion
int ConvertErrorCode(CURLcode curl_code) {
```
Also, make sure that all code aligned with Google C++ code style